src/app/shared/ui/tournament-bracket.component.ts

```diff
@@ -0,0 +1,203 @@
+import {
+  ChangeDetectionStrategy,
+  Component,
+  computed,
+  inject,
+  input,
+} from '@angular/core';
+import { TreeNode } from 'primeng/api';
+import { TreeModule, TreeNodeSelectEvent } from 'primeng/tree';
+import { EventTeamModel } from '../util/supabase-types';
+import { EventService } from '../data-access/event.service';
+import { AvatarModule } from 'primeng/avatar';
+import { AvatarGroupModule } from 'primeng/avatargroup';
+
+interface EventTeamModelWithWinnerFlag extends EventTeamModel {
+  isWinner: boolean;
+}
+
+@Component({
+  selector: 'joshies-tournament-bracket',
+  standalone: true,
+  template: `
+    <p-tree
+      [value]="bracket()"
+      layout="horizontal"
+      [styleClass]="bracket().length ? 'rotate-180' : ''"
+      selectionMode="checkbox"
+      (onNodeSelect)="setMatchWinner($event, bracket(), selectedNodes)"
+      [(selection)]="selectedNodes"
+      propagateSelectionUp="false"
+      propagateSelectionDown="false"
+    >
+      <ng-template let-node pTemplate="node">
+        <div class="flex align-items-center rotate-180 w-15rem h-3rem">
+          <span class="text-sm text-400 mr-1">{{ node.data?.seed }}</span>
+          <p-avatarGroup styleClass="mr-2">
+            @for (
+              participant of node.data.participants;
+              track participant.participant_id
+            ) {
+              <p-avatar
+                [image]="participant.avatar_url"
+                size="large"
+                shape="circle"
+              />
+            }
+          </p-avatarGroup>
+          <span class="text-800">
+            @for (
+              participant of node.data.participants;
+              track participant.participant_id;
+              let index = $index, last = $last
+            ) {
+              {{
+                (last && index !== 0 ? '& ' : '') +
+                  participant.display_name +
+                  (index < node.data.participants.length - 2 ? ',' : '')
+              }}
+            }
```

Comment:
I would convert this logic into a reusable utility function. Then use that utility function in a Pipe so it can be reused in other pages.

Reply:
Created `event-helpers.ts` and `participant-list.pipe.ts` to implement this logic as a pipe.